src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -99,21 +99,29 @@ public void CompileMethod(MethodCodeNode methodCodeNodeNeedingCode)
                     CorJitFlag.CORJIT_FLG_DEBUG_INFO |
                     CorJitFlag.CORJIT_FLG_PREJIT);
 
+                CompilerTypeSystemContext typeSystemContext = _compilation.TypeSystemContext;
+
                 if (!_compilation.Options.NoLineNumbers)
                 {
-                    CompilerTypeSystemContext typeSystemContext = _compilation.TypeSystemContext;
                     IEnumerable<ILSequencePoint> ilSequencePoints = typeSystemContext.GetSequencePointsForMethod(MethodBeingCompiled);
                     if (ilSequencePoints != null)
                     {
-                        Dictionary<int, SequencePoint> sequencePoints = new Dictionary<int, SequencePoint>();
-                        foreach (var point in ilSequencePoints)
-                        {
-                            sequencePoints.Add(point.Offset, new SequencePoint() { Document = point.Document, LineNumber = point.LineNumber });
-                        }
-                        _sequencePoints = sequencePoints;
+                        SetSequencePoints(ilSequencePoints);
                     }
                 }
 
+                IEnumerable<ILLocalVariable> localVariables = typeSystemContext.GetLocalVariableNamesForMethod(MethodBeingCompiled);
```

Comment:
This API was originally just a quick hack for CppCodeGen. It does not return the start/end scopes associated with the local variable since they were not useful for CppCodeGen. This limitation does not exist for RyuJIT/ObjWriter - we should wire the scopes through.